Code/wwlib/mutex.h

```diff
@@ -117,37 +117,21 @@ class CriticalSectionClass
 
 class FastCriticalSectionClass
 {
-	volatile unsigned Flag;
+    std::mutex lock;
```

Comment:
This should really be a std::atomic to get equivalent asm generation, see https://github.com/TheSuperHackers/GeneralsGameCode/blob/main/Core/Libraries/Source/WWVegas/WWLib/mutex.h for an implementation.

Reply:
The name suggests its intended to be fast rather than correct given its name and implementation. Otherwise just replace the use of FastCriticalSectionClass all together rather than making it the same as CriticalSectionClass and drop the duplicate code?